app/code/core/Mage/Adminhtml/Model/System/Config/Backend/File.php

```diff
@@ -144,6 +146,12 @@ protected function _getUploadDir()
      */
     protected function _getUploadRoot($token)
     {
+        $value = Mage::getStoreConfig($token);
+        if (preg_match(self::SYSTEM_FILESYSTEM_REGEX, $value, $matches) !== false) {
+            $dir = str_replace('root_dir', 'base_dir', $matches[1]);
+            $path = str_replace('/', DS, $matches[2]);
+            return Mage::getConfig()->getOptions()->getData($dir) . $path;
+        }
```

Comment:
please revert the changes in this method(and do it as a separate PullRequest with proper description), they are not covered by the Description and there is no reason visible for me, why this change might be needed.

Reply:
I put it in the comment below